### PR TITLE
Change format of add/remove programs registration to improve winget support

### LINKED ISSUE #12469: Winget package manager treats NVDA Alpha as older than version 2020.4
Note, it is likely that we have to file this issue against WinGet, but filing it here as well for reference.

### Steps to reproduce:
1. Install [WinGet](https://docs.microsoft.com/en-us/windows/package-manager/winget/)
2. Execute winget upgrade
3. Note the following output
```
Name    Id                    Version              Available Source
-------------------------------------------------------------------
NVDA    NVAccess.NVDA         alpha-22934,e05715d3 2020.4    winget
```

### Actual behavior:
2020.4 is treated as an update from Alpha, which is certainly not the case.

### Expected behavior:
2020.4 should not be treated as an update from Alpha

#### NVDA installed/portable/running from source:
Installed

#### NVDA version:
alpha-22934,e05715d3

#### Windows version:
Windows 10 21H1

#### Name and version of other software in use when reproducing the issue:
Winget v1.0.11451

### PR DESCRIPTION
**Pull request by LeonarddeR**
### Link to issue number:
Fixes #12469

### Summary of the issue:
Winget is the new package manager in the most recent versions of Windows 10 and 11. It allows automatic updating via the command line. However, Winget threats Alpha versions of NVDA older than any released version. This means that, when an Alpha version for NVDA 2022.3 is installed, Winget will yet threat NVDA 2022.1 as an upgrade. See https://github.com/microsoft/winget-cli/issues/1209#issuecomment-867044235 for more information about version parsing in Winget. In short, our versioning scheme for alpha versions (e.g. `alpha-25875,d6a193ae`) is considered pretty exotic.

I proposed three possible solutions in https://github.com/nvaccess/nvda/issues/12469#issuecomment-867368159. Option 3 is not possible with winget, see https://github.com/microsoft/winget-cli/issues/1209#issuecomment-1185760017. Therefore, this pr chooses option 1, but it extends on that (see below)

### Description of user facing changes
In add/remove programs:

* Old situation:
    - DisplayName: NVDA
    - DisplayVersion: alpha-25875,d6a193ae
* New situation:
    - DisplayName: NVDA alpha-25875,d6a193ae
    - DisplayVersion: '2022.3.0.25875'

In about window:

* Old situation: Version: alpha-25875,d6a193ae
* New situation: Version: alpha-25875,d6a193ae (2022.3.0.25875)

Apart from fixing Winget this way, it will be much easier for end users to see on what year.major.minor branch of NVDA they are with their alpha version.
note that versions in other places are really *not* affected. Therefore this change is barely visible. Also note that having a version in the Displayname and the product version in the DisplayVersion for ARP is pretty common. This is used by Visual Studio for example.

### Testing strategy:
Install a try build and observe that in Windows add/remove programs / apps and features, the pr identifier is moved to the display name whereas the version is constructed of year.major.minor.build.

### Known issues with pull request:
* For released versions, info is slightly doubling in the about window, e.g. Version: 2022.1 (2022.1.0.xxx). We could leave out the doubled parts, but that means we'll lose year.major.minor in Alpha again.

### Change log entries:
Bug fixes
- When updating NVDA using the Windows Package Manager CLI (aka winget), a released version of NVDA is no longer always treated as newer than whatever alpha version is installed. (#12469)

### Code Review Checklist:

- [x] Pull Request description:
  - description is up to date
  - change log entries
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] API is compatible with existing add-ons.
- [x] Documentation:
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English

### CODE REVIEW

**Comment by feerrenrut** [comment]
To progress this, consideration should given to usages of the changed version numbers and what impact it will have.

In particular:
- Is this version number sent to the (NV Access) update check server?
- How will this affect usage statistics gathering?

File: source/installer.py [hunk 1]
```diff
@@ -227,15 +228,15 @@ def removeOldProgramFiles(destPath):
...
-	"InstallDir":u"{installDir}",
-	"Publisher":versionInfo.publisher,
-	"UninstallDirectory":u"{installDir}",
-	"UninstallString":u"{installDir}\\uninstall.exe",
-	"URLInfoAbout":versionInfo.url,
+uninstallerRegInfo: Dict[str, str] = {
+	"DisplayName": f"{versionInfo.name} {versionInfo.version}",
+	"DisplayVersion": versionInfo.version_detailed,
+	"DisplayIcon": os.path.join(installDir, "images", "nvda.ico"),
+	"InstallDir": "{installDir}",
```
Comment (seanbudd) [inline source/installer.py:235]:
```suggestion
	"InstallDir": installDir,
```

File: source/installer.py [hunk 2]
```diff
@@ -227,15 +228,15 @@ def removeOldProgramFiles(destPath):
...
+	"Publisher": versionInfo.publisher,
+	"UninstallDirectory": "{installDir}",
```
Comment (seanbudd) [inline source/installer.py:237]:
```suggestion
	"UninstallDirectory": "installDir",
```

**Comment by leonardder** [comment]
> To progress this, consideration should given to usages of the changed version numbers and what impact it will have.

As noted above, this only has impact to add/remove programs.

> * Is this version number sent to the (NV Access) update check server?

No. versionInfo.version is not changed with this pr.

> * How will this affect usage statistics gathering?

It won't, as that relies on the update check mechanism.

**Review by seanbudd — APPROVED** [review]

**Pull request merged**